src/map/ai/states/ability_state.cpp

```diff
@@ -189,5 +209,43 @@ bool CAbilityState::CanUseAbility()
         }
         return false;
     }
+    else
+    {
+        if (m_PEntity->StatusEffectContainer->HasStatusEffect(EFFECT_AMNESIA) ||
+            (m_PEntity->StatusEffectContainer->HasStatusEffect(EFFECT_IMPAIRMENT) && (m_PEntity->StatusEffectContainer->GetStatusEffect(EFFECT_IMPAIRMENT)->GetPower() == 0x01 || m_PEntity->StatusEffectContainer->GetStatusEffect(EFFECT_IMPAIRMENT)->GetPower() == 0x03)))
+        {
+            return false;
+        }
```

Comment:
This is a bit hard to read, especially because its a combination or AND and ORs. Can you split this out a bit?

```cpp
bool hasAmnesia = ...;
bool hasImpairment = ...;
bool impairmentPower = ...;

if (...)
```

Reply:
Updated

Reply:
👍